lib/ansible/modules/monitoring/zabbix/zabbix_host.py

```diff
@@ -556,15 +556,16 @@ def check_all_properties(self, host_id, group_ids, status, interfaces, template_
                 return True
 
         if inventory_mode:
-            if LooseVersion(self._zbx_api_version) <= LooseVersion('4.4.0'):
-                if host['inventory']:
+            if host['inventory']:
+                # The way to read the inventory_mode is different on versions between [4.0.0, 4.4.0)
+                if LooseVersion(self._zbx_api_version) >= LooseVersion('4.0.0') and LooseVersion(self._zbx_api_version) < LooseVersion('4.4.0'):
```

Comment:
```suggestion
                if LooseVersion(self._zbx_api_version) < LooseVersion('4.4.1'):
```

The following structure has been changed since version 4.4.1.

```
(snip)
    "inventory_mode": "0",
    "inventory": {
      "type": "",
      "type_full": "",
      "name": "",
      "alias": "fuga",
(snip)
```

So, please change the version condition to 4.4.1 or higher.

Reply:
I checked the changelogs for [4.4.1](https://www.zabbix.com/rn/rn4.4.1) and indeed that was the version where it changed. But what about the condition `>= LooseVersion('4.0.0')`? According to the [docs ](https://www.zabbix.com/documentation/3.0/manual/api/reference/host/object) it worked like 4.4.1+

Reply:
Thank you for check Zabbix API documentation and my suggestion.  
I commented about `if` condition, please check it.